### PR TITLE
Code refactor and minor optimize codegeneration compiler

### PR DESCRIPTION
**Pull request by GermanAizek**
@variar, review my changes. I think you will like these minor changes.

### CODE REVIEW

File: src/filewatch/src/filewatcher.cpp [hunk 1]
```diff
@@ -161,23 +161,24 @@ class EfswFileWatcher final : public efsw::FileWatchListener {
...
         auto watchedDirectory
             = std::find_if( watchedPaths_.begin(), watchedPaths_.end(),
                             [ &directory ]( const auto& wd ) { return wd.name == directory; } );
 
         if ( watchedDirectory != watchedPaths_.end() ) {
+            const auto filename = fileInfo.fileName().toStdString();
 
-            auto watchedFile = std::find( watchedDirectory->files.begin(),
-                                          watchedDirectory->files.end(), filename );
+            std::vector<WatchedFile> files = watchedDirectory->files;
```
Comment (variar) [inline src/filewatch/src/filewatcher.cpp:173]: Do we need a copy of files vector here?

**Review by variar — CHANGES_REQUESTED** [review]
Great job! There is only one thing I don’t understand, please see my comment.

File: src/filewatch/src/filewatcher.cpp (hunk 1 above)
Comment (GermanAizek) [reply src/filewatch/src/filewatcher.cpp:173]: @variar, Yes, you noticed the problem correctly, it was more likely needed indicate type `auto&`

Comment (GermanAizek) [reply src/filewatch/src/filewatcher.cpp]: @variar, fixed here: https://github.com/variar/klogg/commit/5d5153ad2c6c5af23e1b27fd179612978aef686b

**Review by variar — APPROVED** [review]

**Comment by variar** [comment]
Many thanks! I'll merge it.

**Pull request merged**